src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -857,7 +857,7 @@ static int client_send_message(sd_dhcp6_client *client, usec_t time_now) {
                                 return r;
                 }
 
-                if (FLAGS_SET(client->request_ia, DHCP6_REQUEST_IA_PD)) {
+                if (FLAGS_SET(client->request_ia, DHCP6_REQUEST_IA_PD) && client->lease->pd.addresses) {
```

Comment:
There's a similar looking check under `DHCP6_STATE_SOLICITATION` that doesn't have `&& client->lease->pd.addresses`. Does it also need to be updated? Admittedly I know very little about this code.

Reply:
I don't know (I don't full understand the code).

For one, that change was not necessary to fix the assertion failures I had in CI, so I didn't do it.
Also, the code there is slightly different, as it passes `&client->ia_pd` instead of `client->lease.pd`.

But yes. Maybe it should!! Dunno.